### ISSUE TITLE
CLI flag to specify `--destBasePath`

### ISSUE THREAD

**Issue by cspotcode** [plopjs/plop#162]
I would like to distribute a portable plopfile via npm.  I know this is a non-standard use-case, but I want to extract generators from my project, publish them to npm, and allow other people within my company to `npx @myemployer/useful-generator` to run the plopfile.  Ideally these users would *not* need to create their own plopfile.js; I realize that a plopfile can declare additional generators by require()ing them, but I'd like to avoid that requirement.

To do this, I roughly need to be able to do the following:

```
plop --cwd $rootOfProject --propfile $rootOfProject/node_modules/@myemployer/useful-generator/reusableplopfile.js --destBasePath $rootOfProject
```

The missing piece is adding a `--destBasePath` flag to plop's CLI.

Unfortunately, using node-plop directly is not ideal.  There's a non-trivial amount of code in [`plop/src/plop.js`](https://github.com/amwmedia/plop/blob/master/src/plop.js) that would need to be duplicated.

**Comment by amwmedia** [comment]
Is the goal here to generate the seed codebase, or to generate structures within an existing codebase? If the goal is to generate structures inside the codebase, I'd highly recommend using [`plop.load()`](https://github.com/amwmedia/plop/blob/master/plop-load.md). This would just require the codebase to have a plopfile with a single line at the top...
```
module.exports = function (plop) {
  plop.load('name-of-my-module');
}
```
This approach gives the codebase the flexibility to grow and expand its use of generators as needed.

If the goal is to generate a seed codebase, I'd recommend adding `process.cwd()` to the data used in the templates (there are several ways to do this) so that all paths can be made relative to the current working directory.

**Comment by cspotcode** [comment]
My use case is a little bit of both categories.  Basically I want to wrap up plop as a CLI tool I can share internally at my company.  So ideally I'd allow both of these usages:
a) expose a CLI tool on the path, such as `companyname-docsite` (`npm install -g @companyname/docsite`)
b) allow people to load it via `plop.load('@companyname/docsite')`

Option (a) could even create the seed plopfile.js.

Would it make a little more sense if this was exposed as a method of plop's API, for example `runPlopFile`?  `await runPlopFile(pathToPlopFile, destBasePath);`

**Comment by amwmedia** [comment]
What would you envision happening when someone calls `runPlopFile()`? A plopfile can have multiple generators inside.

Could you create 2 codebases (1 for the seed, and the other for the project generators)? The seed repo could even use the `create-` convention (`create-docsite`) so someone in your company could call `npm init docsite` to invoke the project scaffolder. That seed project could include a package.json dependency for `@companyname/docsite` and a plopfile that calls `plop.load()` to get it hooked up.

Just a thought.

**Comment by cspotcode** [comment]
I expect runPlopFile to have equivalent behavior to the plop CLI: it will show interactive prompts equivalent to the CLI, asking which generator to run, etc.  It is a JS function-call entrypoint, as opposed to the process-spawning entrypoint offered by the plop CLI.

The following 2 invocations are equivalent:
```bash
# From bash
$ plop --cwd $rootOfProject --plopfile $rootOfProject/node_modules/@myemployer/useful-generator/reusableplopfile.js --destBasePath $rootOfProject
```

```js
// From JS
await require('plop').runPlopFile({
    cwd: rootOfProject,
    plopfile: require.resolve('@myemployer/useful-generator/reusableplopfile.js'),
    destBasePath: rootOfProject
});
```

Splitting into multiple npm modules sounds like extra effort with no benefit; much easier to keep it all in one module.

**Comment by amwmedia** [comment]
released with v2.5.0
